semgrep-core/src/reporting/JSON_report.ml

```diff
@@ -214,6 +214,7 @@ let match_results_of_matches_and_errors files res =
     stats = { okfiles = count_ok; errorfiles = count_errors };
     time = res.RP.rule_profiling |> Common.map_opt json_time_of_profiling_data;
   }
+  |> Semgrep_core_response_util.sort_match_results
```

Comment:
Maybe we should sort the matches instead of the Semgrep_core_respond.match_results?

Reply:
I'm just wondering why we sort so late, why not doing it earlier?

Reply:
It was the easiest way to sort the results and this guarantees that the results are sorted regardless of what produced them. But yes, it's a good idea, in general, to produce results in the least surprising order everywhere possible.